index.d.ts

```diff
@@ -0,0 +1,51 @@
+export interface ActionCreator {
+  (...args: any[]): any;
+}
+
+export type Reducer<S> = (state: S, action: any) => S;
+
+export type Dispatch = (action: any) => any;
```

Comment:
Why isn't this sufficient?
```ts
interface Action {
  type: string;
}
```

If you want to have strict actions you can just do:
```ts
enum ActionType {A, B, C}
interface MyAction<P> {
    type: ActionType;
    payload: P;
}
```
And it will be assignable to `Action`.

Also, reducer should not be constrained to accept only some selected actions. It should accept any possible action and bypass ones it doesn't need to handle.